src/apis/watch.ts

```diff
@@ -329,6 +329,7 @@ function createWatcher(
     // The subsequent callbacks will redirect to `callback`.
     let shiftCallback = (n: any, o: any) => {
       shiftCallback = originalCallback
+      o = isArray(n) ? [] : undefined
       applyCb(n, o)
```

Comment:
I think it's better not to modify the argument
```suggestion
      // o is undefined on the first call
      applyCb(n, isArray(n) ? [] : o)
```